repl/repl.go

```diff
@@ -140,29 +159,28 @@ func EvalString(what string) (string, []string, string) {
 //
 // Options to set AutoLoad and AutoSave and Compact.
 func EvalStringWithOption(o Options, what string) (res string, errs []string, formatted string) {
-	defer func() {
-		if r := recover(); r != nil {
-			errs = append(errs, fmt.Sprintf("panic: %v", r))
-		}
-	}()
 	s := eval.NewState()
+	if o.MaxDepth > 0 {
+		s.MaxDepth = o.MaxDepth
+	}
 	out := &strings.Builder{}
 	s.Out = out
 	s.LogOut = out
 	s.NoLog = true
-	err := AutoLoad(s, o)
-	if err != nil {
-		log.Errf("Error loading autoload file: %v", err)
-	}
-	_, errs, formatted = EvalOne(s, what, out, o)
+	_ = AutoLoad(s, o) // errors already logged
```

Comment:
Why returning an error then?

Reply:
well... still.. maybe there would be a strict mode and you would stop on errors